sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousDataSourceRDD.scala

```diff
@@ -98,6 +98,10 @@ class ContinuousDataSourceRDD(
   override def getPreferredLocations(split: Partition): Seq[String] = {
     split.asInstanceOf[ContinuousDataSourceRDDPartition].inputPartition.preferredLocations()
   }
+
+  override def clearDependencies(): Unit = {
+    throw new IllegalStateException("Continuous RDDs cannot be checkpointed")
```

Comment:
I'm wondering the method can be called in normal situation: when continuous query is terminated.

Reply:
I don't know, I'm unfamiliar with this method. @tdas

Reply:
@HeartSaVioR No this method no intended for being called in normal circumstance. And less of a reason to call this in an internally generated RDD.